### DISCUSSION TITLE
Data parsers + some basic questions

### DISCUSSION THREAD

**Discussion by ligerzero-ai** [materialsvirtuallab/m3gnet#15] [Q&A]
Hi, 

Is there any possibility that the code/scrapers used to parse the VASP output files for the intermediate structures/energies, forces/stresses be provided? i.e. for generating the training data? (laziness on my end)

> "PES data, i.e., intermediate structures and their corresponding energies, forces, stresses, that have been amassed in the process of performing structural relaxations." from the preprint

Also, is there any chance that more extensive tutorials/examples will be written on how to freeze models/output for new users like myself?

Thanks in advance, Han.

**Comment by shyuep** [comment]
The code used for parsing VASP output files is pymatgen. However, we did not actually do that parsing ourselves. The data was directly obtained from the Materials Project itself.

Indeed, we are writing more and more tutorials. Please feel free to let us know what kind of information would be useful.

**Reply by ligerzero-ai** [reply]
Thanks for the fast response - I am mainly interested in the ability to train on my own data - but I guess that it is first necessary to write some OUTCAR scrapers so I can get the relaxation data (did not preserve the vasprun files). I realise pymatgen has an OUTCAR reader but from the documentation I don't think it has the ability to scrape the per-ionic step data out of the box - it probably requires some fiddling with the read_table_pattern and read_pattern functions.

Will drop by with more requests if/when I get around to actually using the package.

**Reply by shyuep** [reply]
Well, yes. We don't usually see a need for OUTCAR parsing because the vasprun is a far better structured document to get most of the info. There are only a few pieces of info that we don't have that we use the OUTCAR parser for.
